src/providers/Wallet.js

```diff
@@ -124,40 +127,25 @@ function WalletAugmented({ children }) {
     [connector]
   )
 
-  const contextValue = useMemo(
-    () => ({
-      ...wallet,
-      connected,
-      handleOnPreferredNetworkChange,
-      preferredNetwork,
-      isSupportedNetwork,
-      resetConnection,
-      ethers,
-    }),
-    [
-      connected,
-      ethers,
-      handleOnPreferredNetworkChange,
-      preferredNetwork,
-      isSupportedNetwork,
-      resetConnection,
-      wallet,
-    ]
-  )
+  // Handle connect automatically if an account is available and we have some connected address on the wallet
+  useEffect(() => connect(), []) // eslint-disable-line
 
-  return (
-    <WalletAugmentedContext.Provider value={contextValue}>
-      {children}
-    </WalletAugmentedContext.Provider>
-  )
-}
+  // This useEffect is needed because we don't have immediately available wallet.chainId right after connecting in the previous hook
+  // We just want to trigger this effect on wallet network change, so we´ll remove preferredNetwork from the useEffect dependencies
+  useEffect(() => {
+    if (wallet.account && preferredNetwork !== wallet.chainId) {
+      if (SUPPORTED_CHAINS.includes(wallet.chainId)) {
+        setPreferredChain(wallet.chainId)
+        setPreferredNetwork(wallet.chainId)
+      }
+    }
+  }, [wallet.account, wallet.chainId]) // eslint-disable-line 
```

Comment:
Removed the preferredNetwork from the effect dependencies to ensure this is only triggered on network change or first render